test/dlc_tests/benchmark/sagemaker/tensorflow/training/test_performance_tensorflow_sm_training.py

```diff
@@ -0,0 +1,56 @@
+import os
+import random
+import re
+import time
+
+import pytest
+
+from invoke.context import Context
+
+from test.test_utils import BENCHMARK_RESULTS_S3_BUCKET
+
+
+@pytest.mark.parametrize("num_nodes", [1, 4], indirect=True)
+def test_tensorflow_sagemaker_training_performance(tensorflow_training, num_nodes, region):
+
+    # This sleep has been inserted because all the parametrized training jobs are automatically created
+    # by SageMaker with the same name, due to being started around the same time, and with the same image uri.
+    time.sleep(random.Random(x=f"{tensorflow_training}{num_nodes}").random() * 60)
+
+    framework_version = re.search(r"[1,2](\.\d+){2}", tensorflow_training).group()
+    processor = "gpu" if "gpu" in tensorflow_training else "cpu"
+
+    ec2_instance_type = "p3.16xlarge" if processor == "gpu" else "c5.18xlarge"
+
+    py_version = "py2" if "py2" in tensorflow_training else "py37" if "py37" in tensorflow_training else "py3"
+
+    time_str = time.strftime('%Y-%m-%d-%H-%M-%S')
+    commit_info = os.getenv("CODEBUILD_RESOLVED_SOURCE_VERSION")
+    target_upload_location = os.path.join(
+        BENCHMARK_RESULTS_S3_BUCKET, "tensorflow", framework_version, "sagemaker", "training", processor, py_version
+    )
+
+    test_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "resources")
+    venv_dir = os.path.join(test_dir, "sm_benchmark_venv")
+
+    ctx = Context()
+
+    with ctx.cd(test_dir), ctx.prefix(f"source {venv_dir}/bin/activate"):
+        log_file = f"results-{commit_info}-{time_str}-{num_nodes}-node.txt"
+        run_out = ctx.run(f"timeout 45m python tf_sm_benchmark.py "
+                          f"--framework-version {framework_version} "
+                          f"--image-uri {tensorflow_training} "
+                          f"--instance-type ml.{ec2_instance_type} "
+                          f"--node-count {num_nodes} "
+                          f"--python {py_version} "
+                          f"--region {region} "
+                          f"> {log_file}",
+                          warn=True, echo=True)
+
+        if not (run_out.ok or run_out.return_code == 124):
+            target_upload_location = os.path.join(target_upload_location, "failure_log")
+
+        ctx.run(f"aws s3 cp {log_file} {os.path.join(target_upload_location, log_file)}")
+        ctx.run(f"cat {log_file}", echo=True)
```

Comment:
This is true. I will make the change to provide just a link to the actual test results. It doesn't look like the last few lines of the file have a lot of useful information about the performance results. We'll need to print about 30 lines of the end of the log file to see useful information.

Reply:
Sorry I did some modification on the comments because I forgot to save it.
We could also consider to do post-process from origin log file to extract the most interesting part to print to codebuild. Of course we might not need to do it.